tests/unit_tests/cpu/core/pdg_particle.cpp

```diff
@@ -0,0 +1,33 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/definitions/pdg_particle.hpp"
+
+// GTest include(s)
+#include <gtest/gtest.h>
+
+using namespace detray;
+
+/// This tests the functionality of a pdg particle
+GTEST_TEST(detray_core, pdg_particle) {
+
+    const auto mu = muon<float>();
```

Comment:
out of curiousity: Does this compile:
```suggestion
    constexpr auto mu = muon<float>();
```

Reply:
ok